### PR TITLE
GET /api/articles (topic query) completed

### PR DESCRIPTION
**Pull request by amedelmouahidi**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -108,6 +108,43 @@ describe("/api/articles", () => {
...
+          });
+        })
+    })
+    test("should respond with 404 when given an invalid topic", () => {
+      return request(app)
+        .get("/api/articles?topic=monalisa")
+        .expect(404)
+        .then((response) => {
+          expect(response.body.msg).toBe("Not found");
+        });
```
Comment (nc-kirsty-butterworth) [inline __tests__/app.test.js:145]: You should have one more test here for a Status 200 - valid topic query but no articles should respond with an empty array of articles 🙂

File: models/articles.models.js [hunk 2]
```diff
@@ -36,23 +36,36 @@ exports.selectArticles = (topic, sort_by = "created_at", order = "desc") => {
...
+      GROUP BY articles.article_id
+      ORDER BY articles.${sort_by} ${order};`;
 
-  queryStr += ` GROUP BY articles.article_id
-  ORDER BY articles.${sort_by} ${order};`;
-  return db.query(queryStr, queryValues).then(({ rows }) => {
-    return rows;
-  });
+  return db.query(queryStr).then(({ rows }) => rows);
 };
```
Comment (nc-kirsty-butterworth) [inline models/articles.models.js:69]: Could this model be DRY'er? Could the queryString be built up if there is a topic instead of having two separate queries to the database depending on the input?

File: __tests__/app.test.js [hunk 3]
```diff
@@ -108,6 +108,43 @@ describe("/api/articles", () => {
         expect(articles).toBeSortedBy("created_at", { descending: true });
       });
   });
+  describe("topic query", () => {
+    test("should respond with an array that filters the articles by the topic value specified in the query", () => {
+      return request(app)
+        .get("/api/articles?topic=mitch")
+        .expect(200)
+        .then(({ body: { articles } }) => {
+          expect(articles.length).toBe(12);
```
[the remark this answers is not in the document]
Comment (amedelmouahidi) [reply __tests__/app.test.js:117]: should I iterate through each artilcle ?

`articles.forEach(article => {
        expect(article.topic).toBe("mitch");
      });`
